contrib/startback-jobs/spec/unit/operation/test_run_job.rb

```diff
@@ -22,16 +18,42 @@ module Jobs
         }).call
       end
 
-      it 'runs the job' do
-        expect(subject).to eql('Hello !!')
+      describe 'When job runs successfuly' do
+
+        let(:job_data) do
+          a_job_data
+        end
+
+        it 'runs the job' do
+          expect(subject).to eql('Hello !!')
+        end
+
+        it 'updates the job' do
+          subject
+          job_info = jobs_relvar.one
+          expect(job_info[:opResult]).to eql('Hello !!')
+          expect(job_info[:isReady]).to eql(true)
+        end
       end
 
-      it 'updates the job' do
-        subject
-        job_info = jobs_relvar.one
-        expect(job_info[:opResult]).to eql('Hello !!')
-        expect(job_info[:isReady]).to eql(true)
+      describe 'When job fails' do
+
+        let(:job_data) do
+          a_job_data({
+            opInput: { 'crash' => true },
+          })
+        end
+
+        it 'updates the job' do
+          subject
+          job_info = jobs_relvar.one
+          puts job_info.inspect
+          expect(job_info[:isReady]).to eql(false)
+          expect(job_info[:opError]).to be_an_instance_of(Startback::Errors::InternalServerError)
```

Comment:
This one won't work with a store that is unable to use marshalling. I wonder whether it would not be safer to save the name of the error instead of the class itself...